java/impl/aws/src/main/java/co/worklytics/psoxy/aws/ParameterStoreConfigService.java

```diff
@@ -134,4 +141,54 @@ String parameterName(ConfigProperty property) {
             return this.namespace + property.name();
         }
     }
+
+    String lockParameterName(String lockId) {
+        return this.namespace + "lock_" + lockId;
+    }
+
+    // make configurable? YAGNI, 2m should be plenty
+    final int LOCK_TTL_SECONDS = 120;
+
+    @Override
+    public boolean acquire(@NonNull String lockId) {
+        final String lockParameterName = lockParameterName(lockId);
+        try {
+            client.putParameter(PutParameterRequest.builder()
+                .name(lockParameterName)
+                .type(ParameterType.STRING)
+                .value("locked")
```

Comment:
if configurable then store the expiration TTL as value, and compare that later?
if value < clock.instant() then release and acquire

Reply:
that requires parsing the the value from `String` to `Instant`, vs just reading `lastModifiedDate` field directly from API